include/xtensor-io/xfile_array.hpp

```diff
@@ -113,7 +114,7 @@ namespace xt
         static constexpr layout_type static_layout = layout_type::dynamic;
         static constexpr bool contiguous_layout = true;
 
-        xfile_array_container() = default;
+        xfile_array_container(const std::string& path, xfile_mode mode=load, value_type init_value=value_type());
```

Comment:
I still don't get the need for an initial value. Initializing with garbage is fine and consistent with in-memory containers of xtensor.